File: nexusarchive-java/src/main/java/com/nexusarchive/entity/EntityConfig.java
// Input: MyBatis-Plus、Lombok、Java 标准库
// Output: EntityConfig 类
// Pos: 领域实体/模型
// 一旦我被更新，务必更新我的开头注释，以及所属的文件夹的 md。

package com.nexusarchive.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 法人配置表
 * 对应表: sys_entity_config
 * 
 * 说明: 每个法人可以有自己的独立配置（ERP接口、业务规则、合规策略等）
 */
@Data
@TableName("sys_entity_config")
public class EntityConfig {

    /**
     * ID
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 法人ID
     */
    private String entityId;

    /**
     * 配置类型: ERP_INTEGRATION, BUSINESS_RULE, COMPLIANCE_POLICY
     */
    private String configType;

    /**
     * 配置键
     */
    private String configKey;

    /**
     * 配置值（JSON格式）
     */
    private String configValue;

    /**
     * 配置描述
     */
    private String description;

    /**
     * 创建人ID
     */
    private String createdBy;

    /**
     * 创建时间
     */
    @TableField(value = "created_time", fill = FieldFill.INSERT)
    private LocalDateTime createdTime;

    /**
     * 更新时间
     */
    @TableField(value = "updated_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedTime;

    /**
     * 逻辑删除
     */
    @TableLogic
    private Integer deleted;
}

File: src/api/entityConfig.ts
// Input: API client 与 ApiResponse
// Output: entityConfigApi
// Pos: 法人配置 API 层
// 一旦我被更新，务必更新我的开头注释，以及所属的文件夹的 md。

import { client } from './client';
import { ApiResponse } from '../types';

/**
 * 法人配置
 */
export interface EntityConfig {
    id?: string;
    entityId: string;
    configType: 'ERP_INTEGRATION' | 'BUSINESS_RULE' | 'COMPLIANCE_POLICY';
    configKey: string;
    configValue?: string; // JSON格式
    description?: string;
    createdBy?: string;
    createdTime?: string;
    updatedTime?: string;
}

export const entityConfigApi = {
    /**
     * 查询指定法人的所有配置
     */
    getConfigsByEntityId: async (entityId: string) => {
        const response = await client.get<ApiResponse<EntityConfig[]>>(
            `/entity-config/entity/${entityId}`
        );
        return response.data;
    },

    /**
     * 查询指定法人和配置类型的配置
     */
    getConfigsByEntityIdAndType: async (entityId: string, configType: string) => {
        const response = await client.get<ApiResponse<EntityConfig[]>>(
            `/entity-config/entity/${entityId}/type/${configType}`
        );
        return response.data;
    },

    /**
     * 查询指定法人的配置（按类型分组）
     */
    getConfigsGroupedByType: async (entityId: string) => {
        const response = await client.get<ApiResponse<Record<string, EntityConfig[]>>>(
            `/entity-config/entity/${entityId}/grouped`
        );
        return response.data;
    },

    /**
     * 保存或更新配置
     */
    saveOrUpdate: async (config: EntityConfig) => {
        const response = await client.post<ApiResponse<{ configId: string }>>(
            '/entity-config',
            config
        );
        return response.data;
    },

    /**
     * 删除指定法人的所有配置
     */
    deleteByEntityId: async (entityId: string) => {
        const response = await client.delete<ApiResponse<void>>(
            `/entity-config/entity/${entityId}`
        );
        return response.data;
    },

    /**
     * 删除指定法人和配置类型的配置
     */
    deleteByEntityIdAndType: async (entityId: string, configType: string) => {
        const response = await client.delete<ApiResponse<void>>(
            `/entity-config/entity/${entityId}/type/${configType}`
        );
        return response.data;
    },
};






